### PR TITLE
Enable casting from Utf8View

### LINKED ISSUE #6076: Support casting to/from Utf8View
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

Continuing the work begun in https://github.com/apache/arrow-rs/issues/5530 to implement casting from Utf8View to all of the types we can cast to from Utf8 and LargeUtf8.

We should be able to cast from Utf8View to other types the same as we can from Utf8 and LargeUtf8 types.

### PR DESCRIPTION
**Pull request by a10y**
# Which issue does this PR close?

Closes #6076 

# Rationale for this change
 
See issue

# What changes are included in this PR?

Mirroring of the existing Utf8 cast operations for Utf8View

# Are there any user-facing changes?

Purely additive, this adds functionality for users that have Utf8View columns.

### CODE REVIEW

File: arrow-cast/src/cast/string.rs [hunk 1]
```diff
@@ -43,17 +44,33 @@ pub(crate) fn parse_string<P: Parser, O: OffsetSizeTrait>(
...
+/// Parse UTF-8 View
+pub(crate) fn parse_string_view<P: Parser>(
+    array: &dyn Array,
+    cast_options: &CastOptions,
+) -> Result<ArrayRef, ArrowError> {
+    let string_view_array = array.as_string_view();
+    parse_string_iter::<P, _, _>(string_view_array.iter(), cast_options, || string_view_array.nulls().cloned())
+}
+
+fn parse_string_iter<'a, P: Parser, I: Iterator<Item=Option<&'a str>>, F: FnOnce() -> Option<NullBuffer>>(
```
Comment (a10y) [inline arrow-cast/src/cast/string.rs:59]: wanted to extract out the common functionality, could also be a macro if preferred

Comment (XiangpengHao) [reply arrow-cast/src/cast/string.rs:59]: Maybe we can use this trait: https://github.com/apache/arrow-rs/blob/master/arrow-string/src/like.rs#L158? Probably need to make it a public trait first.

**Comment by a10y** [comment]
This is ready for review, I've run `cargo test -p arrow-cast` locally and everything passes

**Comment by a10y** [comment]
Hm, I'm not sure what's up with the archery test, I'm seeing some connection denied after 10 minutes, I don't have permission to retrigger the check but I'm wondering if it's possibly a flake?

**Review by alamb — APPROVED** [review]
Thank you @a10y. I took the liberty of running `cargo fmt` and pushing a new commit to this branch to get the CI to pass

**Comment by alamb** [comment]
> Hm, I'm not sure what's up with the archery test, I'm seeing some connection denied after 10 minutes, I don't have permission to retrigger the check but I'm wondering if it's possibly a flake?

Yes, I think it was a flake. Also once we have merged this PR I think your subsequent PRs will automatically start CI so we should be able to reduce the turnaround time on future PRs

**Comment by alamb** [comment]
Thanks again @a10y

**Comment by a10y** [comment]
Thanks for the quick reviews!

**Comment by alamb** [comment]
> Thanks for the quick reviews!

Well, I am not sure I would call 2 days a quick review. One of my goals over the next few months is to grow the reviewer capacity in arrow-rs .

**Pull request merged**